src/RobustLeastSquares.jl

```diff
@@ -1,5 +1,307 @@
+m_estimator_weight(r,est::CauchyEstimator) = 1.0 ./ (1.0 + r.*r/(est.width*est.width))
+m_estimator_sqrtweight(r,est::CauchyEstimator) = 1.0 ./ sqrt(1.0 + r.*r/(est.width*est.width))
+
+
+"""
+A custom type for applying an estimators to only a subset of the range
+(uses L2 for any unspecified range)
+"""
+immutable SubsetEstimatorConvex{T<:MEstimatorConvex} <: MEstimatorConvex
+    rng::UnitRange{Int64}
+    est::T
+end
+immutable SubsetEstimator{T<:MEstimator} <: MEstimatorNonConvex
+    rng::UnitRange{Int64}
+    est::T
+end
+
+function m_estimator_rho{T}(r,est::Union{SubsetEstimatorConvex{T},SubsetEstimator{T}})
+    rho = 0.5*r.^2
+    rho[est.rng] = m_estimator_rho(r[est.rng],est.est)
+    return rho
+end
+function m_estimator_psi{T}(r,est::Union{SubsetEstimatorConvex{T},SubsetEstimator{T}})
+    psi = r
+    psi[est.rng] = m_estimator_psi(r[est.rng],est.est)
+    return psi
+end
+function m_estimator_weight{T}(r,est::Union{SubsetEstimatorConvex{T},SubsetEstimator{T}})
+    w = ones(size(r))
+    w[est.rng] = m_estimator_weight(r[est.rng],est.est)
+    return w
+end
+function m_estimator_sqrtweight{T}(r,est::Union{SubsetEstimatorConvex{T},SubsetEstimator{T}})
+    w = ones(size(r))
+    w[est.rng] = m_estimator_sqrtweight(r[est.rng],est.est)
+    return w
+end
+
+
+"""
+A custom type for applying two estimators to different ranges
+(uses L2 for any unspecified range)
+"""
+immutable DualEstimatorConvex{T1<:MEstimatorConvex,T2<:MEstimatorConvex} <: MEstimatorConvex
+    rng1::UnitRange{Int64}
+    est1::T1
+    rng2::UnitRange{Int64}
+    est2::T2
+end
+immutable DualEstimator{T1<:MEstimator,T2<:MEstimator} <: MEstimatorNonConvex
+    rng1::UnitRange{Int64}
+    est1::T1
+    rng2::UnitRange{Int64}
+    est2::T2
+end
+#DualEstimator = union(DualEstimatorConvex,DualEstimatorNonConvex)
+function m_estimator_rho{T1,T2}(r,est::Union{DualEstimatorConvex{T1,T2},DualEstimator{T1,T2}})
+    rho = 0.5*r.^2
+    rho[est.rng1] = m_estimator_rho(r[est.rng1],est.est1)
+    rho[est.rng2] = m_estimator_rho(r[est.rng2],est.est2)
+    return rho
+end
+function m_estimator_psi{T1,T2}(r,est::Union{DualEstimatorConvex{T1,T2},DualEstimator{T1,T2}})
+    psi = r
+    psi[est.rng1] = m_estimator_psi(r[est.rng1],est.est1)
+    psi[est.rng2] = m_estimator_psi(r[est.rng2],est.est2)
+    return phi
+end
+function m_estimator_weight{T1,T2}(r,est::Union{DualEstimatorConvex{T1,T2},DualEstimator{T1,T2}})
+    w = ones(size(r))
+    w[est.rng1] = m_estimator_weight(r[est.rng1],est.est1)
+    w[est.rng2] = m_estimator_weight(r[est.rng2],est.est2)
+    return w
+end
+function m_estimator_sqrtweight{T1,T2}(r,est::Union{DualEstimatorConvex{T1,T2},DualEstimator{T1,T2}})
+    w = ones(size(r))
+    w[est.rng1] = m_estimator_sqrtweight(r[est.rng1],est.est1)
+    w[est.rng2] = m_estimator_sqrtweight(r[est.rng2],est.est2)
+    return w
+end
+
+"Recreate an M-Estimator of the same type using the MAD (median absolute deviation) of the residual"
```

Comment:
Yeah, you are estimating the scale of the distribution. It would make it feel more like an expectation-maximisation algorithm, i.e. estimate the parameters of the distribution (the scale), calculate the weights, do a weighted regression, repeat, repeat...